File: das_backend/src/main/java/ch/sbb/backend/preload/infrastructure/model/entities/TrainIdentificationEntity.java
package ch.sbb.backend.preload.infrastructure.model.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import lombok.Getter;

@Entity
@Table(name = "train_identification")
@Getter
public class TrainIdentificationEntity {

    @Id
    private Integer id;

    private String trainPathId;

    private Integer period;

    private String operationalTrainNumber;

    private OffsetDateTime startDateTime;

    /**
     * Comma separated list of companies.
     */
    private String companies;

    private LocalDate operationalDay;

}

File: backend/src/main/kotlin/ch/sbb/backend/preload/infrastructure/entities/TrainIdentificationEntity.kt
package ch.sbb.backend.preload.infrastructure.entities

import ch.sbb.backend.preload.domain.TrainIdentification
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.IdClass
import java.time.LocalDate
import java.time.OffsetDateTime

@Entity(name = "train_identification")
@IdClass(TrainIdentificationId::class)
class TrainIdentificationEntity(
    @Id
    var operationalTrainNumber: String,
    @Id
    var startDate: LocalDate,
    @Id
    var company: String,
    var startDateTime: OffsetDateTime
) {
    constructor(trainIdentification: TrainIdentification) : this(
        trainIdentification.operationalTrainNumber,
        trainIdentification.startDate,
        trainIdentification.company,
        trainIdentification.startDateTime
    )
}
